src/services_models/services_constructor.jl

```diff
@@ -23,15 +23,14 @@ end
 
 function construct_service!(
     psi_container::PSIContainer,
-    services::IS.FlattenIteratorWrapper{SR},
+    services::Vector{SR},
     sys::PSY.System,
     model::ServiceModel{SR, RangeReserve},
     devices_template::Dict{Symbol, DeviceModel},
 ) where {SR <: PSY.Reserve}
     services_mapping = PSY.get_contributing_device_mapping(sys)
     time_steps = model_time_steps(psi_container)
-    names = (PSY.get_name(s) for s in services)
-
+    names = [PSY.get_name(s) for s in services]
```

Comment:
@jd-lara Does the content of `names` need to be consistent with the services you add variables and constraints for?  You are calling add_param_container and add_cons_container with names that could later be skipped.

Reply:
No, the validate services already made those checks and only adds the services that are valid.